x-pack/plugin/ccr/qa/multi-cluster/src/test/java/org/elasticsearch/xpack/ccr/AutoFollowIT.java

```diff
@@ -1120,6 +1121,115 @@ public void testAutoFollowSearchableSnapshotsFails() throws Exception {
         }
     }
 
+    public void testNoWarningOnPromoteDatastreamWhenTemplateExistsOnFollower() throws Exception {
+        if ("follow".equals(targetCluster) == false) {
+            return;
+        }
+        testDatastreamPromotionWarnings(true);
+    }
+
+    public void testWarningOnPromoteDatastreamWhenTemplateDoesNotExistsOnFollower() {
+        if ("follow".equals(targetCluster) == false) {
+            return;
+        }
+        WarningFailureException exception = assertThrows(WarningFailureException.class, () -> testDatastreamPromotionWarnings(false));
+        assertThat(
+            exception.getMessage(),
+            containsString(
+                "does not have a matching index template. " + "This will cause rollover to fail until a matching index template is created]"
+            )
+        );
+    }
+
+    private void testDatastreamPromotionWarnings(Boolean createFollowerTemplate) throws Exception {
+        final int numDocs = 64;
+        final String dataStreamName = getTestName().toLowerCase(Locale.ROOT) + "-dopromo";
+        final String autoFollowPatternName = getTestName().toLowerCase(Locale.ROOT);
+
+        int initialNumberOfSuccessfulFollowedIndices = getNumberOfSuccessfulFollowedIndices();
+        List<String> backingIndexNames = null;
+        try {
+            // Create index template
```

Comment:
The comment reffers to the "step" of these test as a whole rather than that line specifically so perhaps:

```suggestion
            // Prepare index template request and execute
```

Reply:
It's ok, you can leave it as is. I think my initial confusion is that this request is sent to both clusters but the execution for the leader cluster was a few lines later, but it's not far enough to warrant a more specific comment.